[BUG]  pytest summary truncates the test failure cause

**Describe the bug**

While trying to create a script counting test failures by cause (with @razajafri ) it appeared that  the lines logged  in the `short test summary info` lack the exception info entirely.  However it turns out it is truncated. Truncation  can be disabled by setting `CI=true` in the environment

**Steps/Code to reproduce bug**
It is particularly visible with any failing test having long decorators   
```
FAILED ../../src/main/python/delta_lake_merge_test.py::test_delta_merge_not_matched_by_source_fallback[DATAGEN_SEED=1738181168, TZ=UTC, IGNORE_ORDER, ALLOW_NON_GPU(ExecutedCommandExec,BroadcastHashJoinExec,ColumnarToRowExec,BroadcastExchangeExec,DataWritingCommandExec,ExecutedCommandExec,DataWritingCommandExec,WriteFilesExec,DeserializeToObjectExec,ShuffleExchangeExec,FileSourceScanExec,FilterExec,MapPartitionsExec,MapElementsExec,ObjectHashAggregateExec,ProjectExec,SerializeFromObjectExec,SortExec)] - pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the pl...
```

**Expected behavior**
A full exception class and message should be visible.

It requires specifying CI=true https://github.com/pytest-dev/pytest/commit/56c266640ef4ee60659f81af36bc95f6e6ca9cfa#diff-351b791015528a514e6df6c55de8f12848e272c705f6cfe2436bbeba4a3204d3R1154



```log
pyspark.errors.exceptions.captured.IllegalArgumentException: Part of the plan is not columnar class com.databricks.sql.transaction.tahoe.constraints.DeltaInvariantCheckerExec
```
**Environment details (please complete the following information)**
 Any

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -17,6 +17,10 @@ set -ex
 SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
 cd "$SCRIPTPATH"
 
+# No failure message truncation in the test summary by default
+# https://github.com/NVIDIA/spark-rapids/issues/12043
+export CI=${CI:-true}
+
 if [[ $( echo ${SKIP_TESTS} | tr [:upper:] [:lower:] ) == "true" ]];
 then
     echo "PYTHON INTEGRATION TESTS SKIPPED..."

